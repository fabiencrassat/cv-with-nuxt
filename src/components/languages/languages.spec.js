import Languages from './languages';
import testUtils from '~~/lib/tests/testUtils';

describe('languages', () => {
  it('mounts properly', () => {
    expect.hasAssertions();
    testUtils.mountProperly(Languages);
  });

  it('renders properly', () => {
    expect.hasAssertions();
    expect(testUtils.htmlFactory(Languages)).toMatchInlineSnapshot(`
      <div>
        <dl></dl>
      </div>
    `);
  });
  it('renders properly with props', () => {
    expect.hasAssertions();
    expect(
      testUtils.htmlFactory(Languages, {
        propsData: {
          languages: {
            french: {
              id: 'french',
              label: 'french',
              description: 'french description',
            },
            english: {
              id: 'english',
              label: 'english',
              description: 'english description',
            },
          },
        },
      })
      // eslint-disable-next-line jest/prefer-inline-snapshots
    ).toMatchSnapshot();
  });
});
