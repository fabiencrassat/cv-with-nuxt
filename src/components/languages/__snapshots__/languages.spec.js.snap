// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languages renders properly with props 1`] = `
<div>
  <dl>
    <div id="french" class="bg-white odd:bg-gray-200 px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
      <dt class="text-sm leading-5 font-medium text-gray-700">
        french
      </dt>
      <dd class="mt-1 text-sm leading-5 text-gray-900 sm:mt-0 sm:col-span-2">
        french description
      </dd>
    </div>
    <div id="english" class="bg-white odd:bg-gray-200 px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
      <dt class="text-sm leading-5 font-medium text-gray-700">
        english
      </dt>
      <dd class="mt-1 text-sm leading-5 text-gray-900 sm:mt-0 sm:col-span-2">
        english description
      </dd>
    </div>
  </dl>
</div>
`;
